Update integration libraries (#13277)
PrefectHQ/prefect: docs/guides/dask-ray-task-runners.md

from prefect import flow, task
from prefect.task_runners import SequentialTaskRunner

@task
def say_hello(name):
    print(f"hello {name}")

@task
def say_goodbye(name):
    print(f"goodbye {name}")

@flow(task_runner=SequentialTaskRunner())
def greetings(names):
    for name in names:
        say_hello.submit(name)
        say_goodbye.submit(name)

if __name__ == "__main__":
    greetings(["arthur", "trillian", "ford", "marvin"])
```

Save this code as `sequential_flow.py` and run it.

<div class="terminal">
```bash
python sequential_flow.py
```
</div>

> If you remove the log messages from the output and just look at the printed output from the tasks, you see they're executed sequentially:

Reviewer: ```suggestion
If you remove the log messages from the output and just look at the printed output from the task runs, you see they're executed sequentially:
```

```
</div>

## Run tasks in parallel with Dask

> You could argue that this basic flow doesn't benefit from parallel execution, but let's proceed so you can see just how simple it is to use the [`DaskTaskRunner`](https://prefecthq.github.io/prefect-dask/).

Reviewer: ```suggestion
This basic flow won't benefit from parallel execution, but let's proceed so you can see just how simple it is to use the [`DaskTaskRunner`](https://prefecthq.github.io/prefect-dask/) for more complex flows.
```

1. Make sure the `prefect-dask` collection is installed by running `pip install -U prefect-dask`.
2. In your flow code, import `DaskTaskRunner` from `prefect_dask.task_runners`.
3. Assign it as the task runner when the flow is defined using the `task_runner=DaskTaskRunner` argument.
4. Use the `.submit` method when calling task-decorated functions.

Example code:

```python hl_lines="2 12 18"
from prefect import flow, task
from prefect_dask.task_runners import DaskTaskRunner

@task
def say_hello(name):
    print(f"hello {name}")

@task
def say_goodbye(name):
    print(f"goodbye {name}")

@flow(task_runner=DaskTaskRunner())
def greetings(names):
    for name in names:
        say_hello.submit(name)
        say_goodbye.submit(name)

if __name__ == "__main__":
    greetings(["arthur", "trillian", "ford", "marvin"])
```

Note that, because you're using `DaskTaskRunner` in a script, you must use `if __name__ == "__main__":` or you'll see warnings and errors.

Run `dask_flow.py`. If you get a warning about accepting incoming network connections, that's okay - everything is local in this example.

<div class="terminal">
```bash
python dask_flow.py
```
</div>

> `DaskTaskRunner` automatically creates a local Dask cluster, then starts executing all of the tasks in parallel. The results do not return in the same order as the sequential code above.

Reviewer: ```suggestion
`DaskTaskRunner` automatically creates a local Dask cluster, then starts executing all of the task runs in parallel. The results do not return in the same order as the sequential code above.
```

    greetings(["arthur", "trillian", "ford", "marvin"])
```

Run the script:

<div class="terminal">
```bash
python dask_flow.py
```
</div>

Once again, the tasks run sequentially. Here's the output with logs removed.

<div class="terminal">
```bash
hello arthur
goodbye arthur
hello trillian
goodbye trillian
hello ford
goodbye ford
hello marvin
goodbye marvin
```
</div>

> The tasks are not submitted to the `DaskTaskRunner`; instead, they run sequentially.

Reviewer: ```suggestion
The task runs are not submitted to the `DaskTaskRunner`; instead, they run sequentially.
```

## Run tasks in parallel with Ray

You can easily switch to Ray as another parallel task runner option.
Use the [`RayTaskRunner`](https://prefecthq.github.io/prefect-ray/) instead of `DaskTaskRunner`.

To configure your flow to use the `RayTaskRunner`:

1. Install `prefect-ray` into your environment with `pip install -U prefect-ray`.
2. In your flow code, import `RayTaskRunner` from `prefect_ray.task_runners`.
3. Specify the task runner when the flow is defined using the `task_runner=RayTaskRunner` argument.

!!! warning "Ray environment limitations"

>     While we're excited about parallel task execution via Ray, there are a few inherent limitations with Ray you should be aware of:

Reviewer: ```suggestion
    While we're excited about parallel task execution via Ray, there are a few limitations with Ray you should be aware of:
```
